solve duplicate dependency issues to unblock pybinding backend migration

Gasoonjia opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11565
* __->__ #11566

This diff creates different namespace for Module and BundledModule under different mode (aten or portable) to hacky solve the duplicated symbol issues in D73564127 since the their are too many complex dependency issues to solve, and lots of users are asking for pybinding backend migration.

In the future this impl should be replaced by shim-like layers.

Differential Revision: [D76405701](https://our.internmc.facebook.com/intern/diff/D76405701/)

Gasoonjia committed:
hacky solution to solve D73564127's duplicate dependency issues to unblock pybinding backend migration

This diff creates different namespace for Module and BundledModule under different mode (aten or portable) to hacky solve the duplicated symbol issues in D73564127 since the their are too many complex dependency issues to solve, and lots of users are asking for pybinding backend migration.

In the future this impl should be replaced by shim-like layers.

Differential Revision: [D76405701](https://our.internmc.facebook.com/intern/diff/D76405701/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "hacky solution to solve D73564127's duplicate dependency issues to unblock pybinding backend migration"


This diff creates different namespace for Module and BundledModule under different mode (aten or portable) to hacky solve the duplicated symbol issues in D73564127 since the their are too many complex dependency issues to solve, and lots of users are asking for pybinding backend migration.

In the future this impl should be replaced by shim-like layers.

Differential Revision: [D76405701](https://our.internmc.facebook.com/intern/diff/D76405701/)

[ghstack-poisoned]

shoumikhin commented:
I vaguely remember a discussion with @swolchok et al. about this hack, did we eventually decide to go with it? Will it be temporary workaround with a timeline to revert when a proper solution is introduced?

Gasoonjia committed:
Update on "solve duplicate dependency issues to unblock pybinding backend migration"


This diff creates different namespace for Module and BundledModule under different mode (aten or portable) to hacky solve the duplicated symbol issues in D73564127 since the their are too many complex dependency issues to solve, and lots of users are asking for pybinding backend migration.

In the future this impl should be replaced by shim-like layers.

Differential Revision: [D76405701](https://our.internmc.facebook.com/intern/diff/D76405701/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "solve duplicate dependency issues to unblock pybinding backend migration"


This diff creates different namespace for Module and BundledModule under different mode (aten or portable) to hacky solve the duplicated symbol issues in D73564127 since the their are too many complex dependency issues to solve, and lots of users are asking for pybinding backend migration.

In the future this impl should be replaced by shim-like layers.

Differential Revision: [D76405701](https://our.internmc.facebook.com/intern/diff/D76405701/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "solve duplicate dependency issues to unblock pybinding backend migration"


This diff creates different namespace for Module and BundledModule under different mode (aten or portable) to hacky solve the duplicated symbol issues in D73564127 since the their are too many complex dependency issues to solve, and lots of users are asking for pybinding backend migration.

In the future this impl should be replaced by shim-like layers.

Differential Revision: [D76405701](https://our.internmc.facebook.com/intern/diff/D76405701/)

[ghstack-poisoned]
